AmplifyTestCommon/Helpers/TypeRegistry.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+@testable import Amplify
+
+extension TypeRegistry: InstanceFactory {}
+
+typealias TypeFactory = (String?) -> Any
+
+class TypeRegistry {
+    enum Failure: Error {
+        case typeNotRegistered(Any.Type)
+    }
+
+    private var registry = [TypeKey: TypeFactory]()
+
+    func register<T>(type: T.Type, factory: @escaping (String?) -> T) {
```

Comment:
does this need to be thread safe?

Reply:
no, it will be called in the unit test once and not during any async activity